File: bussines/internal/models/plan.go
package models

import (
	"time"
)

type Plan struct {
	ID               int       `gorm:"primaryKey" json:"id"`
	Name             string    `gorm:"unique;not null" json:"name"`
	UsersLimit       int       `gorm:"column:usersLimit;default:0" json:"usersLimit"`
	ConnectionsLimit int       `gorm:"column:connectionsLimit;default:0" json:"connectionsLimit"`
	QueuesLimit      int       `gorm:"column:queuesLimit;default:0" json:"queuesLimit"`
	PluginQuota      int       `gorm:"column:pluginQuota;default:0" json:"pluginQuota"`
	Price            float64   `gorm:"type:decimal(10,2)" json:"price"`
	Active           bool      `gorm:"default:true" json:"active"`
	CreatedAt        time.Time `gorm:"column:createdAt" json:"createdAt"`
	UpdatedAt        time.Time `gorm:"column:updatedAt" json:"updatedAt"`
}

func (Plan) TableName() string {
	return "Plans"
}

File: backend/src/models/Plan.ts
import {
  Table,
  Column,
  CreatedAt,
  UpdatedAt,
  Model,
  PrimaryKey,
  AutoIncrement,
  DataType,
  Unique,
  AllowNull
} from "sequelize-typescript";

@Table
class Plan extends Model<Plan> {
  @PrimaryKey
  @AutoIncrement
  @Column
  id: number;

  @Unique
  @AllowNull(false)
  @Column
  name: string;

  @AllowNull(false)
  @Column(DataType.INTEGER)
  pluginQuota: number;

  @AllowNull(false)
  @Column(DataType.DECIMAL(10, 2))
  price: number;

  @AllowNull(false)
  @Column(DataType.BOOLEAN)
  active: boolean;

  @CreatedAt
  createdAt: Date;

  @UpdatedAt
  updatedAt: Date;
}

export default Plan;
